crates/server/src/middlewares/auth.rs

```diff
@@ -0,0 +1,43 @@
+use axum::{
+    extract::{Request, State},
+    http::{HeaderMap, StatusCode},
+    middleware::Next,
+    response::{IntoResponse, Response},
+    Json,
+};
+use db_access::{auth::find_api_key, DbConnection};
+use serde::Serialize;
+
+#[derive(Debug, Serialize)]
+pub struct AuthErrorResponse {
+    error: String,
+}
+
+/// A simple api key auth that checks if the key provided is in the db
+/// TODO: change this to use the more comprehensive tower_http auth middleware.
+pub async fn simple_apikey_auth(
```

Comment:
`axum::extract::Request` struct  has a generic type for the body, add:

```rust
pub async fn simple_apikey_auth<B>(
    State(db): State<DbConnection>,
    headers: HeaderMap,
    request: Request<B>,
    next: Next<B>,
) -> Response
```

 requests might have bodies of different types, such as a String, Bytes, or a deserialized JSON payload

ref https://docs.rs/axum/latest/axum/extract/type.Request.html

Reply:
If you do prefer something a little bit more explicit, I'm ok to change.

Reply:
Yes it defaults to Body, but by just adding a generic we keep our options wider in case in the future we want to change the body's type